### ISSUE TITLE
Don't allow changing class and getting materials at an enemy Outpost / Warboat

### ISSUE THREAD

**Issue by mugg91** [transhumandesign/kag-base#1497]
## Description

It seems odd that you cannot change class at the enemy tent and warboat, but you can change class at the enemy's Outpost. 

It also seems odd that you can get resupplies at the enemies' facilities.
I think that should be something that you can only get at facilities that you actually own.

Therefore this issue entry suggests the following changes:

- Have Outpost only show the class change button to players of the same team.
- Have Outpost give resupplies only to players of the same team.
- Have Warboat give resupplies only to players of the same team.
- Have Tent give resupplies only to players of the same team.

## Additional Information

This is related to `pre 4137 build` PR #1273, which addresses the same core issues.

**Comment by AsuMagic** [comment]
> Have Outpost only show the class change button to players of the same team.

I disagree on this one. I find it interesting from a gameplay/strategical perspective, much like you can use enemy shops to your advantage.

**Comment by mugg91** [comment]
The word of the hour is "consistency". 
Why not show the tunnel button but show the class change button?
Why have enemy tunnels show the tunnel button but not have the enemy outposts do it as well?
Why show a class change button at the enemy outpost but not at the enemy tent, too?
